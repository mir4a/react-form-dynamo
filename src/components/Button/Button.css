.Button {
  font-size: 16px;
  font-weight: 700;
  padding: 5px 25px;
  color: var(--btn-text-color);
  background-color: var(--btn-primary);
  border-color: var(--input-border-color);
  border-radius: var(--input-border-radius);
}


.Button:hover, .Button:focus {
  cursor: pointer;
  background-color: var(--btn-hover);
  box-shadow: 0 2px 10px var(--btn-shadow);
  outline: none;
}

.Button:disabled,
.Button:disabled:hover {
  color: white;
  cursor: initial;
  box-shadow: none;
  background-color: var(--btn-disabled);
}

.Button:active {
  background-color: var(--btn-active);
  outline: none;
}