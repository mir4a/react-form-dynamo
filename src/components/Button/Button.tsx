import React, { memo } from 'react';
import classNames from 'classnames';
import './Button.css';

interface Props {
  label: string;
  onClick(event: React.MouseEvent<HTMLButtonElement>): void;
  className?: string;
  disabled?: boolean;
}

export default memo(function Button(props: Props) {
  const { label, onClick, className, disabled } = props;
  const btnClassNames = classNames('Button', className);

  return (
    <button onClick={onClick} className={btnClassNames} disabled={disabled}>
      {label}
    </button>
  );
});
