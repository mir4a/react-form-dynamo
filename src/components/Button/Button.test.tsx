import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

it('renders correctly', () => {
  const tree = renderer
    .create(<Button label="A" onClick={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders disabled button', () => {
  const tree = renderer
    .create(<Button disabled={true} label="A" onClick={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders button with disabled=false', () => {
  const tree = renderer
    .create(<Button disabled={false} label="A" onClick={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders button with additional class names', () => {
  const tree = renderer
    .create(<Button className="some-css-class" label="A" onClick={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
