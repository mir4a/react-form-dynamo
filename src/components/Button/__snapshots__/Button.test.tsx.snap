// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button with additional class names 1`] = `
<button
  className="Button some-css-class"
  onClick={[Function]}
>
  A
</button>
`;

exports[`renders button with disabled=false 1`] = `
<button
  className="Button"
  disabled={false}
  onClick={[Function]}
>
  A
</button>
`;

exports[`renders correctly 1`] = `
<button
  className="Button"
  onClick={[Function]}
>
  A
</button>
`;

exports[`renders disabled button 1`] = `
<button
  className="Button"
  disabled={true}
  onClick={[Function]}
>
  A
</button>
`;
